File: ambeth/Ambeth.Xml/ambeth/xml/ICyclicXmlHandler.cs
using System;
using System.IO;

namespace De.Osthus.Ambeth.Xml
{
    public interface ICyclicXmlHandler
    {
        String Write(Object obj);

	    void WriteToStream(Stream outputStream, Object obj);

	    Object Read(String cyclicXmlContent);

        Object ReadFromStream(Stream inputStream);
    }
}

File: jambeth/jAmbeth.Xml/src/de/osthus/ambeth/xml/ICyclicXMLHandler.java
package de.osthus.ambeth.xml;

import java.io.InputStream;
import java.io.OutputStream;
import java.nio.channels.ReadableByteChannel;
import java.nio.channels.WritableByteChannel;

public interface ICyclicXMLHandler
{
	String write(Object obj);

	void writeToStream(OutputStream outputStream, Object obj);

	void writeToChannel(WritableByteChannel byteChannel, Object obj);

	Object read(String cyclicXmlContent);

	Object readFromStream(InputStream inputStream);

	Object readFromChannel(ReadableByteChannel byteChannel);
}
